### PR TITLE
feat(form-field): introduce clears api to Form field elements

### PR DESCRIPTION
**Pull request by Theeraphat-Sorasetsakul**
## Description

Introduce Clears to form field.

Fixes # ([DME-2142](https://jira.refinitiv.com/browse/DME-2142))

### Effected elements:
 - Text field
 - Number field
 - Password field
 - Email field
 - Search field
 - Combo box
 - Tree Select
 - Datetime picker

### Effected Change:
- Add a new demo
- Add style for clears
- Code implementation
- Add Tests
- Change content in Text field doc.

### Prefer to not change:
- Doesn't support a11y because Vaadin and native doesn't support it.
- Don't implement in Slider and Select because it looks weird to have clears button and native doesn't support it.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/halo-theme/src/custom-elements/ef-tree-select.less [hunk 1]
```diff
@@ -70,6 +66,11 @@
       color: @input-focused-border-color;
       background: none;
     }
+
```
Comment (wattachai-lseg) [inline packages/halo-theme/src/custom-elements/ef-tree-select.less:65]: This ruleset is unnecessary.
```css
[part~='button']:active {
      color: @button-hover-text-color;
    }
```

Comment (Theeraphat-Sorasetsakul) [reply packages/halo-theme/src/custom-elements/ef-tree-select.less:65]: Fixed.

File: packages/core/src/elements/FormFieldElement.ts [hunk 2]
```diff
@@ -364,4 +382,46 @@ export abstract class FormFieldElement extends ControlElement {
...
+
+  /**
+   * Template for clear button
+   * Rendered when `clears` attribute is set
+   * @returns Popup template or undefined
+   */
+  protected get clearButtonTemplate(): TemplateResult | typeof nothing {
+    if (this.hasClear) {
+      return html`
+        <div
```
Comment (wattachai-lseg) [inline packages/core/src/elements/FormFieldElement.ts:394]: let's add `aria-hidden="true"`.

File: packages/core/src/elements/FormFieldElement.ts [hunk 3]
```diff
@@ -364,4 +382,46 @@ export abstract class FormFieldElement extends ControlElement {
...
+          ${ref(this.clearsButtonRef)}
```
Comment (Theeraphat-Sorasetsakul) [inline packages/core/src/elements/FormFieldElement.ts:396]:
```suggestion
          aria-hidden="true"
          ${ref(this.clearsButtonRef)}
```

File: packages/core/src/elements/FormFieldElement.ts (hunk 2 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/core/src/elements/FormFieldElement.ts:394]: Fixed.

**Review by wattachai-lseg — COMMENTED** [review]
Let's update test snapshots impacted by the new template update.

**Review by wattachai-lseg — APPROVED** [review]

**Comment by bualoy-napat** [comment]
LGTM

**Pull request merged**